rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -1750,10 +1702,9 @@ extern "C"
             {
                 void *data = guard_conditions->guard_conditions[i];
                 GuardCondition *guard_condition = (GuardCondition*)data;
-                if(!guard_condition->getHasTriggered())
-                {
-                    guard_conditions->guard_conditions[i] = 0;
-                }
+                // Guard conditions persist (don't set to null)
+                // We call getHasTriggered to reset the trigger state.
```

Comment:
I thought about it some more and I decided that I agree with Dirk for now.

f8e81aa

I also think that this will work with my current set of changes in rclcpp just fine. Launching CI just in case (I should also probably launch a repeated job).

http://ci.ros2.org/job/ci_osx/895/